edx_when/api.py

```diff
@@ -91,16 +92,20 @@ def get_dates_for_course(course_id, user=None, use_cached=True):
     dates = {}
     policies = {}
     for cdate in qset:
+        if schedule is None:
+            schedule = cdate.schedule_for_user(user)
```

Comment:
Nano-nit: If there aren't valid falsy values for schedule, I generally prefer to tighten this kind of check into one line: `schedule = schedule or cdate.schedule_for_user(user)`  But I get if that style is not everyone's cup of tea.

Reply:
Yeah, I tend to prefer explicitly checking for `None` in cases like this where it's a signal value. But I can see your side of it too.